src/main/java/io/opentracing/contrib/spanmanager/ThreadLocalSpanManager.java

```diff
@@ -0,0 +1,100 @@
+package io.opentracing.contrib.spanmanager;
+
+import io.opentracing.NoopSpan;
+import io.opentracing.Span;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Default {@link SpanManager} implementation using {@link ThreadLocal} storage,
+ * maintaining a stack-like structure of linked managed spans.
+ * <p>
+ * The linked managed spans provide the following stack unwinding algorithm:
+ * <ol>
+ * <li>If the released <em>span</em> is not the managed span, the managed span is left alone.</li>
+ * <li>Otherwise, the first parent that is <em>not yet released</em> is set as the new managed span.</li>
+ * <li>If no managed parents remain, the <em>managed span</em> is cleared.</li>
+ * <li>Consecutive <code>release()</code> calls for already-released spans will be ignored.</li>
+ * </ol>
+ */
+final class ThreadLocalSpanManager implements SpanManager {
```

Comment:
Do you have any practical examples of possible non-TLS SpanManagers that would actually play nice with the static-ness of this API? (I.e., that could retrieve the active span *without parameters* and without using TLS) Or is the idea just that they'd use some *other* TLS, like MDC or similar?